linux/scripts/dist.sh

```diff
@@ -27,13 +32,46 @@ mkdir -p dist
 
 # dist for keyman
 cp -a debian ../
-cd ..
+cd "${KEYMAN_ROOT}"
 echo "3.0 (native)" > debian/source/format
+# shellcheck disable=SC2154
 dch keyman --newversion "${KEYMAN_VERSION}" --force-bad-version --nomultimaint
 
 # Create the tarball
-# Note: the files end up in subdirectories under `keyman`, so we can
-# include that when matching files and directories to ignore.
+
+# shellcheck disable=2034
+to_include=(
+  common/build.sh \
+  common/cpp \
+  common/include \
+  common/linux \
+  common/test/keyboards/baseline \
+  core \
+  linux \
+  resources/build/*.sh \
+  resources/build/meson \
+  resources/standards-data \
+  resources/*.sh \
+  ./*.md \
+  ./build.sh \
+  ./*.json \
+)
+
+# shellcheck disable=2034
+to_exclude=(
+  common/test/keyboards/baseline/kmcomp-*.zip \
+  core/build \
+  linux/build \
+  linux/builddebs \
+  linux/docs/help \
+  linux/keyman-config/keyman_config/version.py \
+  linux/keyman-config/buildtools/build-langtags.py \
+  linux/keyman-system-service/build
+)
+ignored_files=()
+
+generate_tar_ignore_list "./" to_include to_exclude ignored_files "$(basename "${KEYMAN_ROOT}")"
+
 dpkg-source \
   --tar-ignore=*~ \
```

Comment:
Are you planning to put all the `--tar-ignore` entries into `to_exclude` or only folders? If only folders, perhaps a comment to that effect would be helpful?

Reply:
No, those `--tar-ignore`s stay there. `to_exclude` contains the exceptions to the included files/folders in `to_include`, whereas the `--tar-ignore`s here are files/folders that we always want to ignore regardless of their location. Having them here allows us to pass the wildcards to `dpkg-source` - whereas the wildcards in `to_exclude` will be resolved and replaced with multiple `--tar-ignore` entries.

Reply:
That'd be a good code comment!